src/Tizen.NUI.XamlBuild/src/internal/Xaml/MarkupExpressionParser.cs

```diff
@@ -0,0 +1,229 @@
+//
+// MarkupExpressionParser.cs
+//
+// This code is partly salvaged from moonlight. Following licence apply.
+//
+//
+// Author(s):
+//   Moonlight List (moonlight-list@lists.ximian.com)
+//   Stephane Delcroix (stephane@mi8.be)
+//
+// Copyright 2009 Novell, Inc.
+// Copyright 2013 Xamarin, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
```

Comment:
please check the copyright issue.
is it no problem to copy this file and to use it?
also please check other files in this PR.